libsplinter/src/network/peer_manager/interconnect.rs

```diff
@@ -0,0 +1,650 @@
+                            .to_string();
+
+                        // convert connection id to peer id
+                        // if peer id is None, fetch peers to see if they have changed
+                        if peer_id.is_empty() {
+                            *peers = match recv_peer_connector.connection_ids() {
+                                Ok(peers) => peers,
+                                Err(err) => {
+                                    error!("Unable to get peer map: {}", err);
+                                    break;
+                                }
+                            };
+                            peer_id = peers
+                                .get_by_value(connection_id)
+                                .to_owned()
+                                .unwrap_or(&"".to_string())
+                                .to_string();
+                        }
+                        peer_id
+                    };
+
+                    // If we have the peer, pass message to dispatcher, else print error
+                    if !peer_id.is_empty() {
+                        let mut network_msg: NetworkMessage =
+                            match protobuf::parse_from_bytes(&envelope.payload()) {
+                                Ok(msg) => msg,
+                                Err(err) => {
+                                    error!("Unable to dispatch message: {}", err);
+                                    continue;
+                                }
+                            };
+
+                        trace!(
+                            "Received Message from {}: {:?}",
+                            peer_id,
+                            network_msg.get_message_type()
+                        );
+                        match network_dispatcher_sender.send(
+                            network_msg.get_message_type(),
+                            network_msg.take_payload(),
+                            peer_id.to_string(),
+                        ) {
+                            Ok(()) => (),
+                            Err((message_type, _, _)) => {
+                                error!("Unable to dispatch message of type {:?}", message_type)
+                            }
+                        }
+                    } else {
+                        error!("Received message from unknown peer");
+                    }
+                }
+            })
+            .map_err(|err| {
+                PeerInterconnectError::StartUpError(format!(
+                    "Unable to start PeerInterconnect receiver thread {}",
+                    err
+                ))
+            })?;
+
+        let message_sender = self
+            .message_sender
+            .take()
+            .ok_or_else(|| PeerInterconnectError::StartUpError("Already started".to_string()))?;
+        let send_join_handle = thread::Builder::new()
+            .name("Peer Interconnect Sender".into())
```

Comment:
Thread naming consistency

Reply:
Added a story to enforce thread naming format.